bpf/lib/nat.h

```diff
@@ -752,10 +752,15 @@ snat_v4_needs_masquerade(struct __ctx_buff *ctx __maybe_unused,
 
 		target->from_local_endpoint = true;
 
-		err = ct_is_reply4(get_ct_map4(tuple), ctx, l4_off, tuple,
-				   &is_reply);
-		if (IS_ERR(err))
+		err = ct_extract_ports4(ctx, l4_off, CT_EGRESS, tuple, NULL);
+		if (err < 0)
 			return err;
+
+		is_reply = ct_is_reply4(get_ct_map4(tuple), tuple);
+
+		/* SNAT code has its own port extraction logic: */
+		tuple->dport = 0;
```

Comment:
This looks like an extra operation. Isn't `ct_is_reply4` always resetting ports now?

Reply:
I shuffled it out in the last patch - `ct_is_reply()` isn't extracting the ports anymore (that happens in the caller now), so it doesn't reset them either.